### DISCUSSION TITLE
I have some ideas for further tidying up my "text clipboard"

### DISCUSSION THREAD

**Discussion by Raytle** [hluk/CopyQ#1702] [Ideas]
I am grateful for being able to segregate my text and images by using a "text clipboard" and an "images clipboard" as is discussed in [How can I remove all of the images from CopyQ, yet keep all of the text? #1687](https://github.com/hluk/CopyQ/issues/1687#issuecomment-876929136).

Essentially, in my mind, the images no longer clutter up my "main clipboard" which, in my mind, is my "text clipboard."

Be that as it may, I have some ideas for further tidying up my "text clipboard." Would anyone be interested in discussing them? If so, I probably won't respond to anyone until, perhaps, Sunday or Monday.

I perused the docs for CopyQ and initially wondered if [Clipboard Content Visibility](https://copyq.readthedocs.io/en/latest/security.html#clipboard-content-visibility) might do what I have in mind but clearly it does not.

**Comment by hluk** [comment]
One handy command is [Big Data Tab](https://github.com/hluk/copyq-commands/tree/master/Automatic#big-data-tab). It stores larger clipboard data in a separate tab which helps speed up browsing the clipboard tab.

**Comment by Raytle** [comment]
> One handy command is [Big Data Tab](https://github.com/hluk/copyq-commands/tree/master/Automatic#big-data-tab). 



Thanks for letting me know. I appreciate that. That's nice! I had actually thought of that but I hadn't mentioned it to you because I didn't want to overwhelm you.

This is merely my opinion but by default I think CopyQ should have the following five tabs, each with a different background color...

1. Images Tab
2. Text Tab One (for words)
3. Text Tab Two (for sentences)
4. Text Tab Three (for paragraphs)
5. Text Tab Four (for documents the user wants to temporarily backup to CopyQ on the fly)

The user would be able to define the range of characters that would appear beneath a given text tab. I imagine by default that...

1. Text Tab **One** might contain text that has from 1 character to 25 characters, 
2. Text Tab **Two** might contain text that has from 26 characters to 100 characters, 
3. Text Tab **Three** might contain text that has from 101 characters to 500 characters,  and finally
4. Text Tab **Four** might contain text that has from 501 characters or more. 

Allow users to bind a hotkey to search all four text options. In my case I might bind them so that pressing...

1.  `F1` might open Text Tab **One**, 
2. `Shift + F1` might open Text Tab **Two**
3. `Ctrl + Shift + F1` might open Text Tab **Three**
4. `Alt + Ctrl + Shift + F1` might open Text Tab **Four**

In my mind images are distinct from text. Therefore, I might bind `F2` to open my Images Tab.

> It stores larger clipboard data in a separate tab which helps speed up browsing the clipboard tab.

To me the implication is obvious: although ostensibly CopyQ is merely a clipboard manager it's better thought of as a search application which happens to amass data via the clipboard. By applying some rudimentary artificial intelligence it would be possible to help users much more easily find the items they want when searching CopyQ. Put simply: CopyQ would become like a user's personal, desktop version of Google.

I am thinking of applications such as...

1. [fzf](https://github.com/junegunn/fzf)
2. [ranger](https://github.com/ranger)
3. [nnn](https://github.com/jarun/nnn)
3. [lf](https://github.com/gokcehan/lf), and
4.  [broot](https://github.com/Canop/broot).

In particular,  integrating CopyQ with [fzf](https://github.com/junegunn/fzf) seems interesting to me. See, for example, how [Ranger integrates with fzf](https://github.com/ranger/ranger/wiki/Custom-Commands#fzf-integration).
